yocs_joyop/src/joyop.cpp

```diff
@@ -136,12 +157,28 @@ void JoyOp::joyCallback(const sensor_msgs::Joy::ConstPtr& joy)
   deadman_pressed_ = joy->buttons[deadman_button_];
   enable_pressed_ = joy->buttons[enable_button_];
   disable_pressed_ = joy->buttons[disable_button_];
+  magic_pressed_ =  joy->buttons[magic_button_];
+  more_magic_pressed_ =  joy->buttons[more_magic_button_];
 }
 
 void JoyOp::publish()
 {
   boost::mutex::scoped_lock lock(publish_mutex_);
 
+  if ( magic_ != magic_pressed_ ) {
```

Comment:
Shouldn't `magic_pressed_` and `more_magic_pressed_` be initialised in the constructor as well?